### PR TITLE
[node-manager] version map for cluster-autoscaler

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

Made the autoscaler versions correspond to kubernetes versions.

## Why do we need it, and what problem does it solve?

New versions of the autoscaler do not have guaranteed backwards compatibility with older versions of kubernetes.

## Test results
<details><summary>1.23</summary>

  ```shell
root@test-master-0:~# kubectl -n d8-cloud-instance-manager logs cluster-autoscaler-85dcc48f9b-9kp4k cluster-autoscaler | grep "Pod .* is unschedulable" -A7
I0731 11:58:58.610393       1 klogx.go:86] Pod default/test-788dcf6fc8-cfg9v is unschedulable
W0731 11:58:58.610875       1 priority.go:90] Priority expander: node group test-worker-6c5615ed not found in priority expander configuration. The group won't be used.
W0731 11:58:58.610887       1 priority.go:90] Priority expander: no priorities info found for any of the expansion options. No options filtered.
I0731 11:58:58.610895       1 scale_up.go:468] Best option to resize: test-worker-6c5615ed
I0731 11:58:58.610902       1 scale_up.go:472] Estimated 1 nodes needed in test-worker-6c5615ed
I0731 11:58:58.610921       1 scale_up.go:595] Final scale-up plan: [{test-worker-6c5615ed 2->3 (max: 3)}]
I0731 11:58:58.610937       1 scale_up.go:691] Scale-up: setting group test-worker-6c5615ed size to 3
I0731 11:59:07.990907       1 node_instances_cache.go:156] Start refreshing cloud provider node instances cache
root@test-master-0:~# kubectl get no 
NAME                               STATUS   ROLES                  AGE     VERSION
test-master-0                      Ready    control-plane,master   3h5m    v1.23.17
test-worker-6c5615ed-755b4-448v7   Ready    worker                 177m    v1.23.17
test-worker-6c5615ed-755b4-v6967   Ready    worker                 4m17s   v1.23.17
test-worker-6c5615ed-755b4-zffl7   Ready    worker                 40m     v1.23.17
  ```
</details>

<details><summary>1.24</summary>

```shell
root@test-master-0:~# kubectl -n d8-cloud-instance-manager logs -f cluster-autoscaler-6b6bdc574c-djs44 cluster-autoscaler | grep "Pod .* is unschedulable" -A7
I0731 12:36:51.279356       1 klogx.go:86] Pod default/test-846fd4f48-gpl4g is unschedulable
W0731 12:36:51.279894       1 priority.go:90] Priority expander: node group test-worker-42375351 not found in priority expander configuration. The group won't be used.
W0731 12:36:51.279908       1 priority.go:90] Priority expander: no priorities info found for any of the expansion options. No options filtered.
I0731 12:36:51.279930       1 scale_up.go:468] Best option to resize: test-worker-42375351
I0731 12:36:51.279941       1 scale_up.go:472] Estimated 1 nodes needed in test-worker-42375351
I0731 12:36:51.280049       1 scale_up.go:595] Final scale-up plan: [{test-worker-42375351 1->2 (max: 3)}]
I0731 12:36:51.280086       1 scale_up.go:691] Scale-up: setting group test-worker-42375351 size to 2
I0731 12:37:01.295912       1 static_autoscaler.go:341] 1 unregistered nodes present
root@test-master-0:~# kubectl get no
NAME                               STATUS   ROLES                  AGE     VERSION
test-master-0                      Ready    control-plane,master   16m     v1.24.16
test-worker-42375351-6c888-8psrg   Ready    worker                 40s     v1.24.16
test-worker-42375351-6c888-qwm7x   Ready    worker                 8m35s   v1.24.16
```
</details>

<details><summary>1.25</summary>

```shell
root@test-master-0:~# kubectl -n d8-cloud-instance-manager logs -f cluster-autoscaler-5876c8d79c-nzgvj cluster-autoscaler | grep "Pod .* is unschedulable" -A7
I0731 13:08:39.361195       1 klogx.go:86] Pod default/test-59b6656d57-2zbxz is unschedulable
W0731 13:08:39.361739       1 priority.go:90] Priority expander: node group test-worker-ed8472b2 not found in priority expander configuration. The group won't be used.
W0731 13:08:39.361752       1 priority.go:90] Priority expander: no priorities info found for any of the expansion options. No options filtered.
I0731 13:08:39.361761       1 scale_up.go:477] Best option to resize: test-worker-ed8472b2
I0731 13:08:39.361769       1 scale_up.go:481] Estimated 1 nodes needed in test-worker-ed8472b2
I0731 13:08:39.361804       1 scale_up.go:601] Final scale-up plan: [{test-worker-ed8472b2 1->2 (max: 3)}]
I0731 13:08:39.361832       1 scale_up.go:700] Scale-up: setting group test-worker-ed8472b2 size to 2
I0731 13:08:49.416142       1 static_autoscaler.go:349] 1 unregistered nodes present
root@test-master-0:~# kubectl get no 
NAME                               STATUS   ROLES                  AGE   VERSION
test-master-0                      Ready    control-plane,master   18m   v1.25.12
test-worker-ed8472b2-747d7-2m9x4   Ready    worker                 12m   v1.25.12
test-worker-ed8472b2-747d7-9f5fd   Ready    worker                 76s   v1.25.12
```
</details>

<details><summary>1.26</summary>

```shell
root@test-master-0:~# kubectl -n d8-cloud-instance-manager logs -f cluster-autoscaler-6968c5d4f-f2kcl cluster-autoscaler | grep "Pod .* is unschedulable" -A7
I0731 13:40:42.442126       1 klogx.go:87] Pod default/test-d7778b68c-rhl58 is unschedulable
W0731 13:40:42.442982       1 priority.go:90] Priority expander: node group test-worker-22072317 not found in priority expander configuration. The group won't be used.
W0731 13:40:42.443005       1 priority.go:90] Priority expander: no priorities info found for any of the expansion options. No options filtered.
I0731 13:40:42.443487       1 scale_up.go:282] Best option to resize: test-worker-22072317
I0731 13:40:42.443508       1 scale_up.go:286] Estimated 1 nodes needed in test-worker-22072317
I0731 13:40:42.443540       1 scale_up.go:405] Final scale-up plan: [{test-worker-22072317 1->2 (max: 3)}]
I0731 13:40:42.443558       1 scale_up.go:608] Scale-up: setting group test-worker-22072317 size to 2
I0731 13:40:52.461158       1 filter_out_schedulable.go:75] Schedulable pods present
root@test-master-0:~# kubectl get no
NAME                               STATUS   ROLES                  AGE     VERSION
test-master-0                      Ready    control-plane,master   20m     v1.26.7
test-worker-22072317-944c5-fvp9b   Ready    worker                 90s     v1.26.7
test-worker-22072317-944c5-gxg8k   Ready    worker                 12m     v1.26.7
```
</details>

<details><summary>1.27</summary>

```shell
root@test-master-0:~# kubectl -n d8-cloud-instance-manager logs -f cluster-autoscaler-6887b657d8-sd6lw cluster-autoscaler | grep "Pod .* is unschedulable" -A7
I0731 14:16:03.455338       1 klogx.go:87] Pod default/test-8689c4bd8f-28n54 is unschedulable
W0731 14:16:03.455946       1 priority.go:90] Priority expander: node group test-worker-e7a9be9e not found in priority expander configuration. The group won't be used.
W0731 14:16:03.455968       1 priority.go:90] Priority expander: no priorities info found for any of the expansion options. No options filtered.
I0731 14:16:03.455981       1 orchestrator.go:193] Best option to resize: test-worker-e7a9be9e
I0731 14:16:03.455991       1 orchestrator.go:197] Estimated 1 nodes needed in test-worker-e7a9be9e
I0731 14:16:03.456014       1 orchestrator.go:310] Final scale-up plan: [{test-worker-e7a9be9e 1->2 (max: 3)}]
I0731 14:16:03.456032       1 orchestrator.go:582] Scale-up: setting group test-worker-e7a9be9e size to 2
I0731 14:16:13.583681       1 static_autoscaler.go:405] 1 unregistered nodes present
root@test-master-0:~# kubectl get no
NAME                               STATUS   ROLES                  AGE    VERSION
test-master-0                      Ready    control-plane,master   18m    v1.27.4
test-worker-e7a9be9e-67c74-9nggp   Ready    worker                 12m    v1.27.4
test-worker-e7a9be9e-67c74-n8szs   Ready    worker                 4m3s   v1.27.4
```
</details>

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: feature
summary: Made the autoscaler versions correspond to kubernetes versions.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/gcp'...

**Comment by z9r5** [comment]
@vitaliy-sn Rebase please

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/gcp'...

File: candi/version_map.yml [hunk 1]
```diff
@@ -87,6 +87,7 @@ k8s:
     cniVersion: 1.2.0
     bashible: &bashible_k8s_ge_1_23
       !!merge <<: *bashible
+    clusterautoscaler: v1.23.4
```
Comment (RomanenkoDenys) [inline candi/version_map.yml:90]: i think it will be clusterAutoscaler

Comment (RomanenkoDenys) [reply candi/version_map.yml:90]: something like
```
clusterAutoscalerPatch: 3
```
because we have kubernetes version in the yaml key.

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

File: candi/version_map.yml (hunk 1 above)
Comment (vitaliy-sn) [reply candi/version_map.yml]: Done

File: modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{{- range $key, $value := .CandiVersionMap.k8s }}
+  {{- $version := toString $key }}
+  {{- if $value.clusterautoscaler }}
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml:3]: clusterAutoscalerPatch ?

**Review by RomanenkoDenys — COMMENTED** [review]
question

File: modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml (hunk 2 above)
Comment (vitaliy-sn) [reply modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml]: Thanks. Fixed.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

File: candi/version_map.yml [hunk 3]
```diff
@@ -96,6 +100,7 @@ k8s:
     status: preview
     patch: 4
     bashible: *bashible
+    clusterAutoscalerPatch: '0'
```
Comment (RomanenkoDenys) [inline candi/version_map.yml:103]:
```suggestion
    clusterAutoscalerPatch: 0
```

**Review by RomanenkoDenys — COMMENTED** [review]
one nit

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Review by RomanenkoDenys — DISMISSED** [review]
LGTM

File: modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+{{- range $key, $value := .CandiVersionMap.k8s }}
+  {{- $version := toString $key }}
+  {{- if $value.clusterAutoscalerPatch }}
+---
+image: {{ $.ModuleName }}/{{ $.ImageName }}-{{ $version | replace "." "-" }}
+from: {{ $.Images.BASE_ALPINE }}
```
Comment (name212) [inline modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml:6]: use scratch image https://github.com/deckhouse/deckhouse/wiki/How-to-build-minimal-images-from-scratch

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Review by name212 — APPROVED** [review]

**Pull request merged**